DashAI/back/dataloaders/classes/dataloader.py

```diff
@@ -141,41 +146,62 @@ def split_dataset(
                 f"(0 and 1 not included), got {val_size}"
             )
 
-        stratify_column = class_column if stratify else None
+        inputs_columns = dataset["train"].inputs_columns
+        outputs_columns = dataset["train"].outputs_columns
+
+        # Get the number of records
+        n = len(dataset["train"])
+
+        # Generate shuffled indices
+        np.random.seed(seed)
+        indices = np.arange(n)
 
         test_val = test_size + val_size
         val_proportion = test_size / test_val
-        inputs_columns = dataset["train"].inputs_columns
-        outputs_columns = dataset["train"].outputs_columns
-        train_split = dataset["train"].train_test_split(
+
+        # Define stratification array if stratify is True
+        stratify_array = (
+            np.array(dataset["train"][class_column])
+            if stratify and class_column
+            else None
+        )
+
+        # Split the indices
+        train_indices, test_val_indices = train_test_split(
+            indices,
             train_size=train_size,
+            random_state=seed,
+            stratify=stratify_array,
             shuffle=shuffle,
-            seed=seed,
-            stratify_by_column=stratify_column,
         )
-        test_valid_split = train_split["test"].train_test_split(
+        test_indices, val_indices = train_test_split(
+            test_val_indices,
             train_size=val_proportion,
+            random_state=seed,
+            stratify=stratify_array[test_val_indices]
+            if stratify_array is not None
+            else None,
             shuffle=shuffle,
-            seed=seed,
-            stratify_by_column=stratify_column,
         )
-        dataset["train"] = train_split["train"]
-        dataset["test"] = test_valid_split["train"]
-        dataset["validation"] = test_valid_split["test"]
 
-        train_dataset_dict = dataset["train"].to_dict()
-        test_dataset_dict = dataset["test"].to_dict()
-        validation_dataset_dict = dataset["validation"].to_dict()
+        # Convert the indices into boolean masks
+        train_mask = np.isin(np.arange(n), train_indices)
+        test_mask = np.isin(np.arange(n), test_indices)
+        val_mask = np.isin(np.arange(n), val_indices)
```

Comment:
Why not to use `indices` if its the same thing and it was already defined?

```suggestion
        train_mask = np.isin(indices, train_indices)
        test_mask = np.isin(indices, test_indices)
        val_mask = np.isin(indices, val_indices)
```